tortoise/verifying_tortoise.go

```diff
@@ -357,18 +338,18 @@ func (t *turtle) calculateExceptions(layerid types.LayerID, blockid types.BlockI
 	return []map[types.BlockID]struct{}{againstDiff, forDiff, neutralDiff}, nil
 }
 
-func (t *turtle) BlockWeight(voting, voted types.BlockID) int {
+func (t *turtle) BlockWeight(voting, voted types.BlockID) int64 {
 	return 1
 }
 
 // Persist saves the current tortoise state to the database
 func (t *turtle) persist() error {
-	return t.bdp.Persist(mesh.TORTOISE, t)
-}
-
-// RecoverVerifyingTortoise retrieve latest saved tortoise from the database
-func RecoverVerifyingTortoise(mdb retriever) (interface{}, error) {
-	return mdb.Retrieve(mesh.TORTOISE, &turtle{})
+	// TODO(dshulyak) make state eviction a part of regular eviction
+	// HandleIncomingLayer should expose an error for this
+	if err := t.state.Evict(); err != nil {
```

Comment:
why perform eviction inside persist (rather than when verifying a layer, as we do now)?

Reply:
i kept it here because beacause HandleIncomingLayer wasn't exposing error